.github/workflows/freebsd.yml

```diff
@@ -5,22 +5,39 @@ on:
     paths-ignore:
     - '.github/workflows/**'
     - '!.github/workflows/freebsd.yml'
+    - 'doc/**'
+    - 'CONTRIBUTING.md'
+    - 'COPYING'
+    - '.gitignore'
+    - '.editorconfig'
+    - 'NEWS'
+    - 'README.md'
+    - 'style_unify'
   pull_request:
     paths-ignore:
     - '.github/workflows/**'
     - '!.github/workflows/freebsd.yml'
+    - 'doc/**'
+    - 'CONTRIBUTING.md'
+    - 'COPYING'
+    - '.gitignore'
+    - '.editorconfig'
+    - 'NEWS'
+    - 'README.md'
+    - 'style_unify'
 
 jobs:
   freebsd:
     runs-on: macos-12
     steps:
-    - uses: actions/checkout@v3
-    - name: test
+    - name: 'Checkout'
+      uses: actions/checkout@v3
+    - name: 'Install prerequisites and build'
       uses: vmactions/freebsd-vm@v0
       with:
         prepare: |
           pkg install -y meson pkgconf libdrm libXext libXfixes wayland
           pkg install -y -x '^mesa($|-libs)'
         run: |
-          meson _build
-          meson compile -C _build
+          meson setup _build
```

Comment:
I need acked-by from @jbeich who helped to create this .yml and is maintainer for freebsd. Major question I have is: which way better reflects what freebsd is doing in its side? If older way is a better approximation, then I am not ok to do this change even if it follows the spirit of meson.

Reply:
Can you be a little more specific here - are you concerned about the "meson" -> "meson setup" change, "meson compile" -> "meson install" or a mix?

In case you've missed it: running meson without "setup" will flag a warning.

Reply:
These changes aren't even relevant to freebsd specifically. They reflect running Meson on ***any*** operating system.

`meson _build` is implicitly rewritten to `meson setup _build` inside Meson's `main()` function, immediately before argument parsing. It is therefore obviously correct to not rely on that, but to use the official and documented command-line syntax of the `meson` command -- even without considering that Meson now warns you that implicit argument rewriting is deprecated.

Aside for that -- building with Meson is a 3-step process, roughly equivalent to the traditional autotools `./configure && make && make install`. Only for Meson, it is `meson setup _build/` instead of configure, and make invocations are replaced with `ninja`.

In both autotools and meson, running the "install" command will first make sure the build itself is up to date, so ***technically*** you can do this:

```
./configure && make install
```

But people usually don't. Just like they usually build with meson, before installing with meson.

Despite this, the freebsd tests did not test installing at all, only building. The commit message for this change states that the goal of this change is to test installing.

(It does that by collapsing both stages into one, which as I pointed out above, people do not generally do.)

Reply:
Thanks for the comprehensive reply Eli. It's quite reassuring hearing that from someone with deep knowledge of meson's internals.

Reply:
`meson setup` is fine. Likely caused by early examples predating `setup` support (before 0.42.0; `USES=meson` abstraction in FreeBSD appeared during 0.39.1) or based on outdated documentation e.g., https://github.com/mesonbuild/meson/commit/a0ce13d4e9bb

However, I don't like removing `meson compile` in favor of implicit dependency by `meson install`:
- build and install in separate phases match many downstream packaging systems and makes it easy to reason about in logs
- install phase is often a priveleged operation e.g., when installing system-wide for all users, so shouldn't happen by accident/bug during  build
- build may need to be adjusted with arguments not supported by install phase e.g., `--verbose` or `--jobs` (both are often used downstream)